### ISSUE TITLE
Link prediction fails

### ISSUE THREAD

**Issue by vidarmehr** [monarch-initiative/N2V#112]
We merged some branches to develop and now, I get an error when i run link prediction. I work on this issue.

**Comment by callahantiff** [comment]
Do you need some help? What script are you running and what’s the error?

**Comment by vidarmehr** [comment]
Thanks a lot, Tiffany! That would be great if you can help me. If you run runLinkPrediction_ppi.py or runLinkPrediction_with_validation_ppi.py, you get an error like this:

> Traceback (most recent call last):
>   File "/Users/ravanv/PycharmProjects/N2V/runLinkPrediction_ppi.py", line 173, in <module>
>     main(args)
>   File "/Users/ravanv/PycharmProjects/N2V/runLinkPrediction_ppi.py", line 168, in main
>     linkpred(pos_train_graph, pos_test_graph, neg_train_graph, neg_test_graph)
>   File "/Users/ravanv/PycharmProjects/N2V/runLinkPrediction_ppi.py", line 130, in linkpred
>     lp.prepare_labels_test_training()
>   File "/Users/ravanv/PycharmProjects/N2V/xn2v/link_prediction.py", line 57, in prepare_labels_test_training
>     pos_train_edge_embs = self.transform(edge_list=self.pos_train_edges, node2vector_map=self.map_node_vector)
>   File "/Users/ravanv/PycharmProjects/N2V/xn2v/link_prediction.py", line 165, in transform
>     emb1 = node2vector_map[node1]
> KeyError: 'ENSP00000216797'
>

**Comment by vidarmehr** [comment]
set num_steps to 100 that it converges fast:
--num_step 100

**Comment by callahantiff** [comment]
> set num_steps to 100 that it converges fast:
> --num_step 100

Ok, I will take a look now.

**Comment by callahantiff** [comment]
@vidarmehr - OK, got it working!

You should now see the loss print every 10 steps when calling `model.train()`.

If you prefer something else, we can easily change this to different interval, too.

**Comment by vidarmehr** [comment]
Thank you, Tiffany! But, I still get the error message. That is weird!

**Comment by callahantiff** [comment]
The `KeyError: 'ENSP00000216797' error, right? I am also looking at this now. I was fixing a separate issue with the cbow training before. Sorry, I should have been more specific.

**Comment by vidarmehr** [comment]
@callahantiff Yes. No worries. Do not worry about it. It is probably too late to work on this issue now. We can work on that tomorrow.

**Comment by callahantiff** [comment]
Sounds good!

**Comment by vidarmehr** [comment]
@callahantiff Hi Tiffany, I found that read_embeddings() was missing from link_prediction_ppi and link_prediction_with_validation_ppi.  I assume that you had removed read_embedding() because you wanted to use the methods in utils/embedding_utils()? 
I put read_embeddings() back to the link_prediction, but still get the error message. We have a group meeting now, but I will continue working on this issue.

**Comment by callahantiff** [comment]
> @callahantiff Hi Tiffany, I found that read_embeddings() was missing from link_prediction_ppi and link_prediction_with_validation_ppi. I assume that you had removed read_embedding() because you wanted to use the methods in utils/embedding_utils()?
> I put read_embeddings() back to the link_prediction, but still get the error message. We have a group meeting now, but I will continue working on this issue.

Huh, I would have sworn that I had updated that. I'll take a look during your meeting too and see if I can help.

**Comment by callahantiff** [comment]
> @callahantiff Hi Tiffany, I found that read_embeddings() was missing from link_prediction_ppi and link_prediction_with_validation_ppi. I assume that you had removed read_embedding() because you wanted to use the methods in utils/embedding_utils()?
> I put read_embeddings() back to the link_prediction, but still get the error message. We have a group meeting now, but I will continue working on this issue.

Actually, I just look back through develop, and it looks like that function was never in those scripts, rather it was in the `link_prediction.py` script. But, I did totally delete it from there (D'oh!). I am fixing it now. Sorry!! :D

**Comment by callahantiff** [comment]
I updated both scripts to utilize the method, now called `load_embeddings()` from the embedding utils script, replacing the original `read_embeddings()` method. I tested both `link_prediction.py` and `link_prediction_with_validation.py` with this update and they both run fine now. Let me know if you have any trouble.

Sorry I missed that in my PR. I'll be sure to double-check things better moving forward.
